rt/features/logging/src/main/java/org/apache/cxf/ext/logging/event/DefaultLogEventMapper.java

```diff
@@ -125,31 +126,36 @@ private String getPrincipal(Message message) {
         return null;
     }
 
-    private String getJAASPrincipal() {
-        StringBuilder principals = new StringBuilder();
-        Iterator<? extends Object> principalIt = getJAASPrincipals();
-        while (principalIt.hasNext()) {
-            principals.append(principalIt.next());
-            if (principalIt.hasNext()) {
-                principals.append(',');
+    public String getConcatenatedJAASPrincipals() {
+        StringBuilder principalsStringBuilder = new StringBuilder();
+        Set<Principal> principals = getJAASPrincipals();
```

Comment:
I believe we could do:

```
if (principals.isEmpty()){
    return null;
}
```

since `getJAASPrincipals()` may return the empty set which does not need synchronization (moreover, it is backed by static instance).